src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java

```diff
@@ -564,6 +566,40 @@ private String getTrustName(Trust trust) {
     return optionalPlacement;
   }
 
+  private boolean placementHasNewSpecialty(String placementId) {
+    if (placementId != null) {
+      Optional<PlacementSpecialty> placementSpecialty = placementSpecialtyService
+          .findById(placementId);
+      return placementSpecialty.isPresent();
+    }
+    return false;
+  }
+
+  /**
+   * Determine whether the PlacementSpecialty being deleted has been deleted correctly, i.e. the
+   * Placement has also been deleted (isEmpty()) or has been superseded by a new
+   * PlacementSpecialty.
+   *
+   * @param placementId of the PlacementSpecialty to be deleted.
+   * @return true if it deleted correctly according to the conditions above.
+   */
+  public boolean placementSpecialtyDeletedCorrectly(String placementId) {
+    if (placementId != null) {
+      Optional<Placement> placement = placementService.findById(placementId);
+      boolean placementSpecialtyDeletedCorrectly =
+          placement.isEmpty() || placementHasNewSpecialty(placementId);
+      if (!placementSpecialtyDeletedCorrectly) {
+        enrich(placement.get());
```

Comment:
It believe it shouldn't throw an Exception, because if `placement.isEmpty()` then boolean `placementSpecialtyDeletedCorrectly` is true, and `!placementSpecialtyDeletedCorrectly` would be false, so it wouldn't go inside the _if_ statement.

There's a test checking that the check is being performed in the `onAfterDelete `method: [L65 - PlacementSpecialtyEventListenerTest](https://github.com/Health-Education-England/tis-trainee-sync/pull/103/files#diff-7eaad9cac503f04b534e3570c51ba6322326b79b69a8fe9290402cee068f1449R65)

I added three more tests in PlacementEnricherFacadeTest to account for all three possible scenarios prompted by the check in `onAfterDelete`:
- test that if `placement.isEmpty()` then do nothing (deletion was good, so no re-enriching or re-requesting)
- test that if `!placement.isEmpty()` and it has a new `placementSpecialty` then do nothing (deletion was good, so no re-enriching or re-requestin)
- test that if `placement.isEmpty()` && there's no `placementSpecialty` linked to it, then something might have gone wrong so do `enrich(placement)` which will end up re`requesting(placementSpecialty)`.